### DISCUSSION TITLE
Simple set label property in code does not work

### DISCUSSION THREAD

**Discussion by FraserElectronics** [AvaloniaUI/Avalonia#11648] [General]
I have just started playing with AvaloniaUI and have stumbled at the first hurdle.
I have a UserControl with a button and a label. 
In the button click event handler I set the labels IsVisible property to true.

The XAML code:

```
<StackPanel Orientation="Vertical">
        <Button Width="300" Height="100" Margin="20" Click="OnButtonClick">
            <Panel HorizontalAlignment="Center" VerticalAlignment="Center">
                <TextBlock>Click Me</TextBlock>
            </Panel>
        </Button>
        <Label x:Name="LblLabel" IsVisible="False">Hello, you clicked me !</Label>
    </StackPanel>
```

The code behind:

```
public partial class MainView : UserControl
{
    public MainView()
    {
        InitializeComponent();
    }

    private void InitializeComponent()
    {
        AvaloniaXamlLoader.Load(this);
    }

    private void OnButtonClick(object sender, RoutedEventArgs e)
    {
        LblLabel.IsVisible = true;
    }
```

When I run this and click on the button, the LblLabel is NULL and so it crashes.
This is being compiled and run under the latest Ubuntu.
When I duplicate this code in Visual Studio under Windows, it works as I would expect.

What am I doing wrong?

Andy

**Comment by rabbitism** [comment]
Which version are you playing with? 
You should make your code like this. `InitlaizeComponent` will be automatically generated. 
```csharp
public partial class MainView : UserControl
{
    public MainView()
    {
        InitializeComponent();
    }

    private void OnButtonClick(object sender, RoutedEventArgs e)
    {
        LblLabel.IsVisible = true;
    }
```

**Reply by FraserElectronics** [reply]
@rabbitism I am using JetBrains Rider 2023.1.2 and that is the initialization code it created.
I have removed the ```InitializeComponent``` method and it is now working fine.
Thanks for you help.

Andy

**Comment by timunie** [comment]
Just to mention: IMO the easiest way for bigger Apps is to work with the MVVM pattern. If you are not familiar with those, you can check out our MVVM samples: https://github.com/AvaloniaUI/Avalonia.Samples#mvvm-samples

**Comment by FraserElectronics** [comment]
Thanks, I am well experienced in MVVM - I just went for a simple project just to test out how much AvaloniaUI differs from WinUI and WPF.

Andy
